### PR TITLE
Tweaks the blastcannon range a bit and makes it care about walls.

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About The Pull Request
I was looking at explosion code recently and noticed that explosions had a way for objects to decrease the range of explosions passing through them. I've applied that code for the blastcannon so now walls will actually do something to reduce it's power. I've also unified the scaling factors for the base effective ranges of the blastwave to compensate for the diminishing effect of walls.

## Why It's Good For The Game
The blastcannon is supposed to be a straight-line explosion. Explosions care about walls. This brings the blastwave into line with normal explosions.
The blastcannon ranges now more accurately match their goal of being proportional to the bombs theoretical effective area.

## Changelog

:cl:
balance: The blastcannon's damage has become more spread out and it now may be somewhat diminished by walls depending on config.
balance: The base range of the blastcannon has been slightly buffed to compensate for the new diminishing effect of walls.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/special/blastcannon.dm [hunk 1]
```diff
@@ -304,37 +300,51 @@
...
-	heavy_ex_range = max(heavy_ex_range - 1, 0)
-	medium_ex_range = max(medium_ex_range - 1, 0)
-	light_ex_range = max(light_ex_range - 1, 0)
-
-	if(heavy_ex_range)
-		SSexplosions.highturf += loc
+	var/decrement = 1
+	var/atom/location = loc
+	if (reactionary)
+		var/atom/location = loc
```
Comment (SuperNovaa41) [inline code/modules/projectiles/guns/special/blastcannon.dm:322]: double definition

Comment (TemporalOroboros) [reply code/modules/projectiles/guns/special/blastcannon.dm:322]: Yeah. I noticed. I'll fix it when I'm done testing a fix for the bug mentioned in #63859.

Comment (TemporalOroboros) [reply code/modules/projectiles/guns/special/blastcannon.dm]: Done.

**Comment by Mosslick** [comment]
So its just a nerf for already niche traitor item nobody uses?

**Comment by Shadowflame909** [comment]
A nerf to an item i've never seen used. Ever.

Explosions are kinda hard to make. And they do better not being straight line only for the real "make the station unusable damage"

THUS! I say this nerf is misguided, and only further disincentives the use of an item that collects dust, and instead promotes the easier and more gratifying massive station hull breach!

**Comment by TemporalOroboros** [comment]
Eh. You're right. I'll buff it a bit to compensate.

**Comment by TemporalOroboros** [comment]
@Mosslick @Shadowflame909 
I have slightly buffed the base range to compensate for the effect of walls.

**Comment by TemporalOroboros** [comment]
@Sealed101 What was your concern?

**Comment by cacogen** [comment]
The blast cannon creates ugly strips of space tiles that just look like someone deleted them instead of leaving broken tiles in its wake like regular explosions. It also causes every area it passes through to be vented, nullifying firelocks and creating stationwide breaches of the type that usually go unfixed. So I, for one, like the idea of lessening its impact a bit by making it respect walls like apparently regular explosions do.

**Pull request merged**